### ISSUE TITLE
[TRI-1739] Vercel 308 redirect causes endpoint to not be reachable

### ISSUE THREAD

**Issue by ericallam** [triggerdotdev/trigger.dev#786]
When prod/staging endpoints are configured with a root domain (e.g. `https://example.com/api/trigger`), and a redirect is configured in vercel to redirect `example.com` -> `www.example.com`, we fail to follow the redirect and are unable to connect to the endpoint. We should handle this situation and make sure redirects are followed (and better yet, update the endpoint to the correct URL, since the redirect is permanent (308)

<sub>[TRI-1739](https://linear.app/triggerdotdev/issue/TRI-1739/vercel-308-redirect-causes-endpoint-to-not-be-reachable)</sub>

**Comment by glowinginthedark** [comment]
Hey @ericallam, I've attempted to re-create this issue and what I've found is that when we configure `example.com` as a staging endpoint and try to run a job on it while it [308] redirects to `www.example.com` (where the actual service is running), the run stays queued and fails with the error: `Connection could not be established to the endpoint`. If we configure the endpoint directly to be `www.exmaple.com`, then the run succeeds.

Does this match your observations?
